src/mongocxx/read_preference.hpp

```diff
@@ -108,6 +110,7 @@ class MONGOCXX_API read_preference {
     /// @deprecated The constructor with no arguments and the method mode() should be used.
     ///
     MONGOCXX_DEPRECATED read_preference(read_mode mode);
+    read_preference(read_mode mode, deprecated_tag);
```

Comment:
Do I understand correctly that you added an overloaded method with an explicit new argument to force this version over the MONGOCXX_DEPRECATED variant above?

Reply:
This is a constructor, not a method, and I'm not forcing anything. I'm providing a way that users can continue to call the deprecated method without getting deprecation warnings/errors, but they need to explicitly write out 'deprecated_tag', so it is clear at the call site that they are using somethign deprecated. I can't use the same trick I used elsewhere of just appending `_deprecated` to the method name, because the name of a constructor must match the class name.